common/src/main/java/com/wynntils/handlers/scoreboard/ScoreboardSegment.java

```diff
@@ -4,88 +4,65 @@
  */
 package com.wynntils.handlers.scoreboard;
 
-import com.wynntils.handlers.scoreboard.type.SegmentMatcher;
-import java.util.ArrayList;
 import java.util.List;
+import java.util.Objects;
+import org.apache.commons.lang3.ArrayUtils;
 
 public final class ScoreboardSegment {
-    private final SegmentMatcher matcher;
+    private final ScoreboardPart scoreboardPart;
     private final String header;
-    private final int startIndex;
+    private List<String> content;
+    private boolean visible = true;
 
-    private String end;
-    private List<String> content = null;
-    private int endIndex = -1;
-    private boolean changed;
-
-    public ScoreboardSegment(SegmentMatcher matcher, String header, int startIndex) {
-        this.matcher = matcher;
+    public ScoreboardSegment(ScoreboardPart scoreboardPart, String header, List<String> content) {
+        this.scoreboardPart = scoreboardPart;
         this.header = header;
-        this.startIndex = startIndex;
-        this.changed = false;
-    }
-
-    @Override
-    public String toString() {
-        return "Segment[" + "matcher="
-                + matcher + ", " + "header="
-                + header + ", " + "content="
-                + content + ", " + "startIndex="
-                + startIndex + ", " + "endIndex="
-                + endIndex + ']';
-    }
-
-    public List<String> getContent() {
-        return content;
-    }
-
-    public int getStartIndex() {
-        return startIndex;
-    }
-
-    public int getEndIndex() {
-        return endIndex;
+        this.content = content;
     }
 
-    public SegmentMatcher getMatcher() {
-        return matcher;
+    public ScoreboardPart getScoreboardPart() {
+        return scoreboardPart;
     }
 
     public String getHeader() {
         return header;
     }
 
-    public String getEnd() {
-        return end;
-    }
-
-    public List<String> getScoreboardLines() {
-        List<String> lines = new ArrayList<>(this.content);
-        lines.add(this.header);
-        if (this.end != null) {
-            lines.add(this.end);
-        }
-
-        return lines;
+    public List<String> getContent() {
+        return content;
     }
 
-    public boolean isChanged() {
-        return changed;
+    public boolean isVisible() {
+        return visible;
     }
 
-    public void setContent(List<String> content) {
-        this.content = content;
+    public void setVisibility(boolean visible) {
```

Comment:
Just checking here. Had it not been for visible, this could have been an immutable record?

Reply:
Probably. Does it really change anything?

Reply:
No, I guess not. It's nice to have immutable holder classes as records, if possible. When a situation occurs that a class is mostly immutable but has a mutable field, it is worth thinking once more if that field actually belongs in the class. In this case I think it's fine. 

(As an example of the opposite is QuestInfo, which should really be split into two parts -- an immutable info about the quest per se, and one mutable QuestProgress part. I have a branch doing this but it got stuck and has probably bit-rotted beyond repair by now)